src/LaunchDarkly/FeatureFlag.php

```diff
@@ -77,8 +83,17 @@ public function isOn() {
      */
     public function evaluate($user, $featureRequester) {
         $prereqEvents = array();
-        $value = $this->_evaluate($user, $featureRequester, $prereqEvents);
-        return new EvalResult($value, $prereqEvents);
+        if (is_null($user) || strlen($user->getKey()) === 0) {
```

Comment:
This is the same move that I made in Java- all flag eval logic now happens in the FeatureFlag class.